modules/lib/lib-admin/src/main/resources/lib/xp/admin.ts

```diff
@@ -13,16 +13,34 @@ declare global {
  * @module admin
  */
 
-const i18n = require('/lib/xp/i18n');
-const portal = require('/lib/xp/portal');
+interface ApiUrlParams {
+    application: string;
+    api?: string;
+    type?: 'server' | 'absolute' | 'websocket';
+    params?: object;
+    path?: string | string[];
+}
+
+interface Portal {
```

Comment:
Why not use the types from the source, which is available in the same code repo?